src/components/SimpleSlider.js

```diff
@@ -0,0 +1,174 @@
+    onPercentChange: React.PropTypes.func.isRequired,
+    percent: React.PropTypes.number.isRequired,
+    selectedColor: React.PropTypes.string,
+    styles: React.PropTypes.object
+  },
+
+  getDefaultProps () {
+    return {
+      disabled: false,
+      selectedColor: StyleConstants.Colors.PRIMARY
+    };
+  },
+
+  getInitialState () {
+    return {
+      dragging: false,
+      leftPixels: 0,
+      width: 0
+    };
+  },
+
+  componentDidMount () {
+    this._setDefaults();
+  },
+
+  componentWillReceiveProps (newProps) {
+    if (this.props.percent !== newProps.percent) {
+      const leftPixels = (newProps.percent * this.state.width);
+
+      this.setState({ leftPixels });
+    }
+  },
+
+  _setDefaults () {
+    const component = ReactDOM.findDOMNode(this.rangeSelectorRef);
+    const width = component.clientWidth;
+    const leftPixels = this.props.percent * width;
+
+    this.setState({ width, leftPixels });
+  },
+
+  _handleMouseEvents (e) {
+    const clientX = e.touches ? e.touches[0].clientX : e.clientX;
+    const leftSpace = ReactDOM.findDOMNode(this.rangeSelectorRef).getBoundingClientRect().left;
+    let currentPercent = ((clientX - leftSpace) / this.state.width);
+
+    if (currentPercent < 0) {
+      currentPercent = 0;
+    } else if (currentPercent > 1) {
+      currentPercent = 1;
+    }
+
+    this.props.onPercentChange(currentPercent);
+  },
+
+  _handleDragStart () {
+    this.setState({
+      dragging: true
+    });
+  },
+
+  _handleDragging (e) {
+    if (this.state.dragging) {
+      this._handleMouseEvents(e);
+    }
+  },
+
+  _handleDragEnd () {
+    this.setState({
+      dragging: false
+    });
+  },
+
+  render () {
+    const styles = this.styles();
+    const { disabled } = this.props;
+
+    return (
+      <div style={styles.component}>
+        <div
+          onMouseLeave={disabled ? null : this._handleDragEnd}
+          onMouseMove={disabled ? null : this._handleDragging}
+          onMouseUp={disabled ? null : this._handleDragEnd}
+          onTouchEnd={disabled ? null : this._handleDragEnd}
+          onTouchMove={disabled ? null : this._handleDragging}
+          ref={(ref) => {
+            this.rangeSelectorRef = ref;
+          }}
+          style={styles.range}
+        >
+          <div
+            onMouseDown={disabled ? null : this._handleMouseEvents}
+            style={styles.trackHolder}
+          >
+            <div style={styles.track} />
+            <div style={styles.selected} />
+          </div>
+          <div
+            onMouseDown={disabled ? null : this._handleDragStart}
+            onTouchStart={disabled ? null : this._handleDragStart}
+            style={styles.toggle}
+          />
+        </div>
+      </div>
+    );
+  },
+
+  styles () {
+    let cursorStyle = 'grab';
+
+    if (this.props.disabled) {
+      cursorStyle = 'not-allowed';
+    }
+    if (this.state.dragging) {
+      cursorStyle = 'grabbing';
+    }
+
+    return _merge({}, {
```

Comment:
I tired `Object.assign` the problem is that it doesn't do a deep merge, if you update 1 thing it just wipes out all the rest.